packages/maskbook/src/web3/UI/EthereumIfERC20TokenApproved.tsx

```diff
@@ -0,0 +1,121 @@
+import { createStyles, Grid, makeStyles } from '@material-ui/core'
+import BigNumber from 'bignumber.js'
+import { useSnackbar } from 'notistack'
+import { useCallback, useEffect } from 'react'
+import ActionButton from '../../extension/options-page/DashboardComponents/ActionButton'
+import { formatBalance } from '../../plugins/Wallet/formatter'
+import { useI18N } from '../../utils/i18n-next-ui'
+import { unreachable } from '../../utils/utils'
+import { ApproveState, useERC20TokenApproveCallback } from '../hooks/useERC20TokenApproveCallbackV2'
+import { TransactionStateType } from '../hooks/useTransactionState'
+import type { ERC20TokenDetailed } from '../types'
+
+const useStyles = makeStyles((theme) =>
+    createStyles({
+        button: {},
+    }),
+)
+
+export interface EthereumIfERC20TokenApprovedProps {
+    amount: string
+    spender: string
+    token?: ERC20TokenDetailed
+    children?: React.ReactNode
+}
+
+export function EthereumIfERC20TokenApproved(props: EthereumIfERC20TokenApprovedProps) {
```

Comment:
I suggest to rename it to `EthereumERC20TokenApprovedBoundary` (component form) or make it a hook that returns JSX children. This suggestion also applies to another component

```js
useEnsureEthereumERC20TokenApproved(amount: string, spender: string, token?: T, children?: ReactNode)
```

Reply:
It's done!